features/0454-present-proof-v2/README.md

```diff
@@ -0,0 +1,289 @@
+
+## Version Change Log
+
+### 2.0 - Alignment with [RFC 0434 Issue Credential](../0453-issue-credential-v2/README.md)
+
+- The "formats" field is added to all the messages to link the specific attachment IDs with the verifiable presentation format and version of the attachment.
+- The details that are part of each message type about the different attachment formats serves as a registry of the known formats and versions.
+- Version 2.0 uses &lt;angle brackets&gt; explicitly to mark all values that may vary between instances, such as identifiers and comments.
+
+## Summary
+
+A protocol supporting a general purpose verifiable presentation exchange regardless of the specifics of the underlying verifiable presentation request and verifiable presentation format.
+
+## Motivation
+
+We need a standard protocol for a verifier to request a presentation from a prover, and for the prover to respond by presenting a proof to the verifier. We doing that exchange, we want to provide a mechanism for the participants to negotiate the underlying format and content of the proof.
+
+## Tutorial
+
+### Name and Version
+
+`present-proof`, version 2.0
+
+### Key Concepts
+
+This protocol is about the messages to support the presentation of verifiable claims, not about the specifics of particular verifiable presentation formats. [DIDComm attachments](../../concepts/0017-attachments/README.md) are deliberately used in messages to make the protocol agnostic to specific verifiable presentation format payloads. Links are provided in the message data element descriptions to details of specific verifiable presentation implementation data structures.
+
+Diagrams in this protocol were made in draw.io. To make changes:
+
+- upload the drawing HTML from this folder to the [draw.io](https://draw.io) site (Import From...GitHub),
+- make changes,
+- export the picture and HTML to your local copy of this repo, and
+- submit a pull request.
+
+### Roles
+
+The roles are `verifier` and `prover`.  The `verifier` requests the presentation of a proof and verifies the presentation, while the `prover` prepares the proof and presents it to the verifier. Optionally, although unlikely from a business sense, the `prover` may initiate an instance of the protocol using the `propose-presentation` message.
+
+### States
+
+#### States for Verifier
+
+* request-sent
+* proposal-received
+* presentation-received
+* abandoned
+* done
+
+#### States for Prover
+
+* request-received
+* proposal-sent
+* presentation-sent
+* abandoned
+* done
+
+For the most part, these states map onto the transitions shown in the choreography diagram in obvious ways. However, a few subtleties are worth highlighting:
+
+* The final transitions for the verifier (Send Presentation Reject, Send Presentation Ack, and Receive Presentation Reject) all result in a final `done` state, but this `done` may or may not be the verifiers intended outcome. We may want to tweak this in a future version of the protocol.
```

Comment:
> Send Presentation Reject, Send Presentation Ack, and Receive Presentation Reject

What is the difference between "send presentation reject" and "receive presentation reject"? The mismatch of terms between the diagram and this sentence is confusing and makes it hard to compare the two.

Does "send presentation reject" refer to the verifier's transition "send request-presentation" -> "receive problem-report"?

Does "receive presentation reject" refer to a transition not illustrated in the diagram where the presentation is received and yet is not valid?

Reply:
I didn't play with that section in V2 and will have another look.

Reply:
As I interpret this, the "send presentation reject" is sent by the verifier and "receive presentation reject" is received by the verifier.  I'll reword to make that clear and that those send the state to "abandoned", not "done" as a terminal state.

Ah...I see.  You are thinking of "receive presentation reject" as another "receive presentation AND reject"?  Which I think is "send presentation reject".  Sigh... :-)

Reply:
@swcurran I think this section (including the choreography diagram) needs cleanup. State transitions are critical for a protocol RFC, and there seem to be a few missing here as we've discussed above.

The existing diagram might get too cluttered if you add them. Consider perhaps switching to a state matrix.